### PR TITLE
Add `diff` command

### LINKED ISSUE #1376: add support for `diff` on running container
### What is the problem you're trying to solve

Please add support for `docker diff` to inspect changes to files or directories on a running container

### Describe the solution you'd like

```
$ nerdctl diff <container>
C /etc
D /etc/profile
A /foo
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by minuk-dev**
#1376 

- Add `diff` command:
  - Use `continuity/fs` to compare filesystem.

- I referred to https://github.com/containerd/nerdctl/blob/e674fe7ba6e49f12e88cd9c6c442e7ea5232502c/pkg/imgutil/commit/commit.go#L73-L95 :
  - But, I don't have any idea to test it. If you have it, please let me know. :bow:

- (Updated) I found the test of `diff` command only works on linux, so remove its test from windows. I think I don't use a platform specific feature, but the test output is weird.:
  - Q. Can I make an issue for windows and try in another pr? I'm not sure I can make it.

### CODE REVIEW

File: cmd/nerdctl/container_diff.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	parent, err := sn.View(ctx, randomID, rootfsID)
+	if err != nil {
+		return nil, err
+	}
+	defer sn.Remove(ctx, randomID)
+
+	var changes []fs.Change
+	err = mount.WithReadonlyTempMount(ctx, parent, func(lower string) error {
+		return mount.WithReadonlyTempMount(ctx, mounts, func(upper string) error {
+			fs.Changes(ctx, lower, upper, func(ck fs.ChangeKind, s string, fi os.FileInfo, err error) error {
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_diff.go:177]: The return value from fs.Changes shouldn’t be ignored

File: cmd/nerdctl/container_diff.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		changes = append(changes, fs.Change{
+			Kind: fs.ChangeKindModify,
+			Path: filepath.Join(newDirPath[:i+1]...),
+		})
+	}
+	return append(changes, new)
+}
+
+func diffShellComplete(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	// show container names (TODO: only show containers with logs)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_diff.go:223]: TODO seems wrong

File: cmd/nerdctl/container_diff.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+			}
+
+			for _, change := range changes {
+				switch change.Kind {
+				case fs.ChangeKindAdd:
+					fmt.Fprintln(cmd.OutOrStdout(), "A", change.Path)
+				case fs.ChangeKindModify:
+					fmt.Fprintln(cmd.OutOrStdout(), "C", change.Path)
+				case fs.ChangeKindDelete:
+					fmt.Fprintln(cmd.OutOrStdout(), "D", change.Path)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_diff.go:97]: OutOrStdout should be called only on e

File: cmd/nerdctl/main.go [hunk 4]
```diff
@@ -251,6 +251,7 @@ Config file ($NERDCTL_TOML): %s
 		newPortCommand(),
 		newStopCommand(),
 		newStartCommand(),
+		newDiffCommand(),
```
Comment (AkihiroSuda) [inline cmd/nerdctl/main.go:254]: Please update https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md too

File: cmd/nerdctl/container_diff.go (hunk 1 above)
Comment (minuk-dev) [reply cmd/nerdctl/container_diff.go]: https://github.com/containerd/nerdctl/pull/2626/commits/00c0d3b726b46caa95618efc76bdc4ffb1814493
Thanks a lot. :bow:

File: cmd/nerdctl/container_diff.go (hunk 2 above)
Comment (minuk-dev) [reply cmd/nerdctl/container_diff.go]: Fixed at https://github.com/containerd/nerdctl/pull/2626/commits/d03a5993400c400a84f9e2c5c2d96d4f736db193

🙇‍♂️

File: cmd/nerdctl/container_diff.go (hunk 3 above)
Comment (minuk-dev) [reply cmd/nerdctl/container_diff.go]: Fixed at https://github.com/containerd/nerdctl/pull/2626/commits/3d1a5f1157be20bcd801181d1582e8edc074df60
Thanks a lot really. 👍

File: cmd/nerdctl/main.go (hunk 4 above)
Comment (minuk-dev) [reply cmd/nerdctl/main.go:254]: Added at https://github.com/containerd/nerdctl/pull/2626/commits/a78b50efbf1100f8a395818e6854ec73a04ebf9e

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by AkihiroSuda** [comment]
Next time please squash commits

**Pull request merged**